src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -789,51 +844,88 @@ async function webviewPreloads(style: PreloadStyles, rendererData: readonly Webv
 		}
 	});
 
-	interface MarkupRenderer {
-		renderMarkup: (context: { element: HTMLElement, content: string }) => void;
+	interface RendererApi {
+		renderCell: (id: string, context: ICreateCellInfo) => void;
+	}
+
+	class Renderer {
+		constructor(
+			public readonly data: RendererMetadata,
+			private readonly loadDependency: (id: string) => Promise<void>,
+		) { }
+
+		private _loadPromise: Promise<RendererApi> | undefined;
+		private _api: RendererApi | undefined;
+
+		public get api() { return this._api; }
+
+		public load(): Promise<RendererApi | undefined> {
+			if (!this._loadPromise) {
+				this._loadPromise = Promise.all(this.data.dependencies.map(dependencyId => this.loadDependency(dependencyId)))
```

Comment:
we discussed it, but noting so it doesn't get forgotten: this is conflating the kernel `dependencies` with renderer dependencies. When this is changed to an `extends` model it should be solved.